tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -28,10 +28,8 @@
 #include "utils/s2n_safety.h"
 
 static const float minimum_send_percent = 5.0;
-static const uint32_t max_client_run_time = 300;
```

Comment:
At some point in the PR that added the timeout there was a comment explaining it: https://github.com/wcs1only/s2n/blob/92a1598bb62cbbae5fb620c38739acb046a8e910/tests/unit/s2n_self_talk_nonblocking_test.c#L42-L45

Is the server never finishing its send a concern at all? Is there any reason to fix the macro instead of removing it?

Reply:
Ya, I saw that comment, but this check never worked before the PR, and it hasn't ever worked since the PR. So I would say that the server never finishing is not a concern, or at least we haven't ever seen it.

This check adds a decent bit of complexity to the flow of this test, so I would say that it's not a "free" check. Plus I wouldn't want each test to be responsible for handling potential deadlocks. If we're worried about that case I think we should set a timeout for ctest, rather than handling this `n` number of times for each suspect test.